### PR TITLE
Update docs, experimental and samples

### PR DESCRIPTION
**Pull request by marbre**
Updates the documentation, scripts in `experimental` and the samples to reflect the renaming of the Python packages.

### CODE REVIEW

File: docs/website/docs/reference/bindings/python.md
Comment (ScottTodd) [inline docs/website/docs/reference/bindings/python.md:1]: Should add a note about the package renaming to this page. I can help with the text for that if you want.

**Review by ScottTodd — COMMENTED** [review]
LGTM once we

1. land the version change
2. land the package name change
3. publish our first nightly and stable releases of the new packages

File: docs/website/docs/reference/bindings/python.md
Comment (marbre) [reply docs/website/docs/reference/bindings/python.md:1]: I can do it tomorrow during my work hours if you don't get to it. That will be still before the regular schedules nightlies are build.

**Comment by ScottTodd** [comment]
> LGTM once we
> 
>     1. land the version change
> 
>     2. land the package name change
> 
>     3. publish our first nightly and stable releases of the new packages

These steps are complete. Can you sync this?

File: docs/website/docs/reference/bindings/python.md [hunk 1]
```diff
@@ -15,15 +15,22 @@ components:
...
 
 Collectively, these packages allow for importing from frontends, compiling
 towards various targets, and executing compiled code on IREE's backends.
 
+???+ Note "Note - `iree-compiler` and `iree-runtime` are deprecated"
+    The Python packages `iree-compiler` and `iree-runtime` have been
+    renamed to `iree-base-compiler` and `iree-base-runtime`
+    respectively, thus effectively deprecating the old packages.
+    This name change only affects the names of the packages but not
+    the modules.
```
Comment (ScottTodd) [inline docs/website/docs/reference/bindings/python.md]: Depending on how pip behaves, we may want a tip to `pip uninstall iree-compiler iree-runtime` during the transition period (I just start with fresh venvs :P)

**Review by ScottTodd — APPROVED** [review]

File: docs/website/docs/reference/bindings/python.md (hunk 1 above)
Comment (marbre) [reply docs/website/docs/reference/bindings/python.md]: Do you prefer to put in in the note above or to https://github.com/iree-org/iree/blob/main/docs/website/docs/guides/deployment-configurations/snippets/_iree-compiler-from-release.md and https://github.com/iree-org/iree/blob/main/docs/website/docs/guides/deployment-configurations/snippets/_iree-runtime-from-release.md ?

Comment (ScottTodd) [reply docs/website/docs/reference/bindings/python.md]: I think it's fine on the Python bindings page, not the snippet that gets included everywhere. If we get user reports showing symptoms of multiple conflicting packages being installed then we can put it in the snippet too.

Comment (marbre) [reply docs/website/docs/reference/bindings/python.md]: Added a note. Let me know if you think this is ready to land.

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**